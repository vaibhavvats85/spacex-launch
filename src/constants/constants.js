export const APP_TITLE = 'SpaceX Launch Programs';
export const DEVELOPER_NAME = 'Vaibhav Vats';
export const DEVELOPED_BY = 'Developed by';
export const MISSION_IDS = 'Mission Ids';
export const LAUNCH_YEAR = 'Launch Year';
export const SUCCESSFULL_LAUNCH = 'Successful Launch';
export const SUCCESSFULL_LANDING = 'Successful Landing';
export const FILTERS = 'Filters';
export const LAUNCH_ERROR = 'Unable to Load Launch Programs. We will get bakc to you soon.';
export const NO_PROGRAM = 'No programs available for the given Criteria';
export const LAUNCH_YEAR_KEY = 'launch_year';
export const LAUNCH_SUCCESS_KEY = 'launch_success';
export const LAND_SUCCESS_KEY = 'land_success';
