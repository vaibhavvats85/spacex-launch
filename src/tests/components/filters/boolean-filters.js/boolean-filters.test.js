import { shallow } from 'enzyme';
import { useLocation } from 'react-router-dom';
import BooleanFilters from '../../../../components/filters/type/boolean-filter';
import * as constants from '../../../../constants';
let wrapper;
let props = {
    type: constants.LAUNCH_SUCCESS_KEY,
    clickEvent: jest.fn()
}
jest.mock('react-router-dom', () => ({
    useLocation: () => ({ search: '?launch_success=true' }),
}));

beforeEach(() => {
    wrapper = shallow(<BooleanFilters {...props} />);
});

test('should render LaunchYearFilters component correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should check button correctly', () => {
    const active = wrapper.find('Button').at(0).prop('active').toLowerCase();
    const id = wrapper.find('Button').at(0).prop('value').id.toLowerCase();
    expect(id).toEqual(active)
});


test('should call handleClick event', () => {
    wrapper.find('Button').at(0).prop('handleClick')();
    expect(props.clickEvent).toHaveBeenCalled();
});
