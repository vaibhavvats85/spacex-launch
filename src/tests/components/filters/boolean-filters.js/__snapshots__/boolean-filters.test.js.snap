// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LaunchYearFilters component correctly 1`] = `
Array [
  <Button
    active="truelaunch_success"
    group="launch_success"
    handleClick={[Function]}
    key="0"
    value={
      Object {
        "id": "Truelaunch_success",
        "val": "True",
      }
    }
  />,
  <Button
    active="truelaunch_success"
    group="launch_success"
    handleClick={[Function]}
    key="1"
    value={
      Object {
        "id": "Falselaunch_success",
        "val": "False",
      }
    }
  />,
]
`;
